securitysystem/pantilt.py

```diff
@@ -0,0 +1,195 @@
+        GPIO.setup(self.__tiltpin,GPIO.OUT)
+        #define PWM servos
+        self.__panservo = GPIO.PWM(self.__panpin,PULSE_HZ)
+        self.__tiltservo = GPIO.PWM(self.__tiltpin,PULSE_HZ)
+
+    def start_servo(self):
+        """
+        Start PWM running on both servos, value of 0 (pulse off) 
+        """
+        logging.info('Start Pan/Tilt Servos')
+        self.__panservo.start(0)
+        self.__tiltservo.start(0)
+
+    def change_pan_angle(self):
+        """
+        Changes angle of pan servo according to user input
+        Parameters
+        ----------
+        angleInDegrees : int
+            Angle to turn to in degrees
+        """
+        #Ask user for angle
+        while True: 
+            angle = float(input('PAN: Enter angle between 0 & 180 (or -1 to EXIT): '))
+            if (angle == -1):
+                logging.info('Pan Angle Not Set')
+                return ANGLE_NOT_SET
+            if (angle >= 0) and (angle <= 180):
+                break
+            logging.info('ERROR: angle is not between 0 & 180')
+
+        #Turn servo to angle
+        self.__panservo.ChangeDutyCycle(2+(angle/18))
+        sleep(WAIT_TIME_SECS)
+        self.__panservo.ChangeDutyCycle(0)
+        logging.info('Pan Angle Set')
+        return ANGLE_SET
+
+    def change_tilt_angle(self):
+        """
+        Changes angle of tilt servo according to user input
+        """
+        #Ask user for angle
+        while True: 
+            angle = float(input('TILT: Enter angle between 0 & 180 (or -1 to EXIT): '))
+            if (angle == -1):
+                logging.info('Tilt Angle Not Set')
+                return ANGLE_NOT_SET
+            if (angle >= 0) and (angle <= 180):
+                break
+            logging.info('ERROR: angle is not between 0 & 180')
+
+        #Turn servo to angle
+        self.__tiltservo.ChangeDutyCycle(2+(angle/18))
+        sleep(WAIT_TIME_SECS)
+        self.__tiltservo.ChangeDutyCycle(0)
+        logging.info('Tilt Angle Set')
+        return ANGLE_SET
+
+    def close_servo(self):
+        """
+        Stops PWM on both servos
+        """
+        logging.info('Stop Pan/Tilt Servos')
+        self.__panservo.stop()
+        self.__tiltservo.stop() 
+
+
+def pantilt_test():
+    """
+    Creates an PanTilt object for manual verification
+    """
+    try:
+        servo_test = PanTilt()
+        servo_test.start_servo()
+        
+        while POLLING:
+            pan_result = servo_test.change_pan_angle()
+            if(pan_result):
+                sleep(WAIT_TIME_SECS)
+
+            tilt_result = servo_test.change_tilt_angle()
+            if(tilt_result):
+                sleep(WAIT_TIME_SECS)
+                
+    except KeyboardInterrupt:
+        logging.info('Exiting')
+    except BaseException as e:
+        logging.error('An error or exception occurred: ' + str(e))
+    finally:
+        servo_test.close_servo()
+        GPIO.cleanup()
+        
+
+# def parse_args():
```

Comment:
Are these lines going to be removed or updated later?

Reply:
im not sure yet but most likely removed

Reply:
removed